src/kafka/producer/kafka_producer.cc

```diff
@@ -0,0 +1,127 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Copyright (C) 2019 ScyllaDB Ltd.
+ */
+
+#include <sstream>
+#include <vector>
+#include <iostream>
+
+#include "../protocol/kafka_primitives.hh"
+#include "../protocol/metadata_request.hh"
+#include "../protocol/metadata_response.hh"
+#include "../protocol/api_versions_request.hh"
+#include "../protocol/api_versions_response.hh"
+#include "../connection/tcp_connection.hh"
+#include "../protocol/produce_request.hh"
+#include "../protocol/produce_response.hh"
+
+#include <boost/iostreams/device/back_inserter.hpp>
+#include <boost/iostreams/stream.hpp>
+#include <boost/iostreams/device/array.hpp>
+
+#include <seastar/core/print.hh>
+#include <seastar/kafka/producer/kafka_producer.hh>
+
+namespace seastar {
+
+namespace kafka {
+
+kafka_producer::kafka_producer(std::string client_id) {
+    _correlation_id = 0;
+    _client_id = client_id;
+}
```

Comment:
This should be done with an initializer list. I did move it in my connection pool PR, but probably better to not merge this at all and correct now

Reply:
And moving the string ofc